libopenage/engine.cpp

```diff
@@ -548,4 +542,12 @@ void Engine::move_phys_camera(float x, float y, float amount) {
 	this->engine_coord_data->camgame_phys += cam_delta.to_phys3();
 }
 
+void Engine::start_game(std::unique_ptr<GameMain> game) {
+	// TODO: maybe implement a proper 1-to-1 connection
+	assert(game);
```

Comment:
There are also some crazy checks that can't be in release. So, I didn't invent any macro for them and used the assert from the library.

I don't see exceptions as fatal errors, so haven't considered them to be the right tool. If there is a explicitly defined fatal exception that no one would try to start handling later in the project, then it'll be usable.

Another concern is the `libopenage/gui/guisys`, which is actually a future qt-sdl library. It should stay as Qt: exception-neutral.